### PR TITLE
[Distributed][Macro] Handle more cases in distributed protocol macro

### PR DESCRIPTION
**Pull request by ktoso**
The distributed protocol macro was blocked from generating the full right thing until we fixed generic actors and protocols and the Codable actor behaviors.

A few more cases to be handled here.

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
Lots of cleanups to the macros, thanks to landing all our previous changes 👍

File: lib/Sema/TypeCheckDeclPrimary.cpp [hunk 1]
```diff
@@ -3945,8 +3944,7 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
 
     checkExplicitAvailability(ED);
 
-    if (nominal->isDistributedActor())
-      TypeChecker::checkDistributedActor(SF, nominal);
+    TypeChecker::checkDistributedActor(SF, nominal);
```
Comment (ktoso) [inline lib/Sema/TypeCheckDeclPrimary.cpp:3947]: moved the if into the func for cleaner `primary` functions

File: test/Distributed/Macros/distributed_macro_expansion_DistributedProtocol_various_requirements.swift [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+// RUN: %empty-directory(%t-scratch)
+
+// RUN: %target-swift-frontend-emit-module -emit-module-path %t/FakeDistributedActorSystems.swiftmodule -module-name FakeDistributedActorSystems -disable-availability-checking %S/../Inputs/FakeDistributedActorSystems.swift
+// RUN: %target-swift-frontend -typecheck -verify -disable-availability-checking -plugin-path %swift-plugin-dir -parse-as-library -I %t %S/../Inputs/FakeDistributedActorSystems.swift -dump-macro-expansions %s -dump-macro-expansions 2>&1 | %FileCheck %s
+
+import Distributed
+
+@_DistributedProtocol
+protocol Greeter: DistributedActor where ActorSystem == FakeActorSystem {
+  distributed func greet(name: String) -> String
```
Comment (xedin) [inline test/Distributed/Macros/distributed_macro_expansion_DistributedProtocol_various_requirements.swift:17]: I'm just going to leave an enhancement request here - I think we need to get tested at least:

- methods with generic parameters
- distributed variables i.e. protocol requirements like `distributed var compute: Int { get set }`
- Multiple distributed members

File: test/Distributed/Runtime/distributed_actor_localSystem_generic.swift [hunk 3]
```diff
@@ -17,7 +17,6 @@
 import Distributed
 
 @available(SwiftStdlib 6.0, *)
-//distributed actor Worker<ActorSystem> where ActorSystem: DistributedActorSystem<any Codable>, ActorSystem.ActorID: Codable {
```
Comment (xedin) [inline test/Distributed/Runtime/distributed_actor_localSystem_generic.swift:20]: Leftover comment

**Review by xedin — APPROVED** [review]

File: test/Distributed/Macros/distributed_macro_expansion_DistributedProtocol_various_requirements.swift (hunk 2 above)
Comment (ktoso) [reply test/Distributed/Macros/distributed_macro_expansion_DistributedProtocol_various_requirements.swift:17]: Sounds good, will do

Comment (ktoso) [reply test/Distributed/Macros/distributed_macro_expansion_DistributedProtocol_various_requirements.swift:17]: Good call on properties actually, I don't think we roundtrip tested them yet

File: test/Distributed/Runtime/distributed_actor_localSystem_generic.swift (hunk 3 above)
Comment (ktoso) [reply test/Distributed/Runtime/distributed_actor_localSystem_generic.swift:20]: that's deleting it, yeap :)

**Comment by ktoso** [comment]
@swift-ci please smoke test

File: test/Distributed/Macros/distributed_macro_expansion_DistributedProtocol_various_requirements.swift (hunk 2 above)
Comment (ktoso) [reply test/Distributed/Macros/distributed_macro_expansion_DistributedProtocol_various_requirements.swift:17]: Covered all these and will add runtime tests as well

**Comment by ktoso** [comment]
@swift-ci please smoke test Windows

File: test/Distributed/Macros/distributed_macro_expansion_DistributedProtocol_various_requirements.swift (hunk 2 above)
Comment (xedin) [reply test/Distributed/Macros/distributed_macro_expansion_DistributedProtocol_various_requirements.swift:17]: Sounds good!

**Pull request merged**